core/src/no/ntnu/tdt4240/asteroids/game/effect/MultishotEffect.java

```diff
@@ -31,12 +31,11 @@ protected TextureRegion getEffectTexture() {
     @Override
     protected void applyEffect(PooledEngine engine, Entity entity, EffectComponent effectComponent) {
         shootComponent = shootMapper.get(entity);
-        oldHandler = shootComponent.handler;
         shootComponent.handler = new MultiShotHandler(800, 3, 5);
     }
 
     @Override
     protected void removeEffect(PooledEngine engine, Entity entity, EffectComponent effectComponent) {
-        shootComponent.handler = oldHandler;
+        shootComponent.handler = new StandardShotHandler();
```

Comment:
This makes it difficult and confusing to modify the default handler because it has to be modified in multiple places